### ISSUE TITLE
Previous links still rendering on graph growth

### ISSUE THREAD

**Issue by n3a9** [vasturiano/react-force-graph#238]
**Describe the bug**

My application adds to the graph and rerenders it with new nodes and links progressively. On every new render for the graph,  the graph canvas still renders the links from all previous graphs. This leads the appearance of random links in the space.

**To Reproduce**

[codesandbox](https://codesandbox.io/s/determined-burnell-0mvme?fontsize=14&hidenavigation=1&theme=dark)

**Expected behavior**

The graph shouldn't render the previous links.

**Comment by vasturiano** [comment]
@n3a9 thanks for reaching out.

This generally happens when you mutate your data structure and the link objects are referencing node objects that are not present in your new nodes structure. Did you check if this is the case?

**Comment by n3a9** [comment]
Yes, I've made sure that all the referenced nodes in the edges exist in the node structure of the graph.

Specifically, currently the `links` portion of the graph is an array of `{to: x, from: y}` objects, and the `nodes` portion is an array of strings. I confirmed that for every `x` and `y` in the links exist in the new `nodes` array (the updates only push to the previous array).

I think that the links that are not updating are from the previous array. So for example, if we have `{nodes: ['x', 'y'], links: [{to: x, from: y}]}` and it grows to `{nodes: ['x', 'y', 'z'], links: [{to: x, from: y}, {to: x, from: y}]}` then the previous x-y edge will become disconnected.

**Comment by vasturiano** [comment]
@n3a9 in your sandbox example the `graph` data structure is empty, is that intended?
```js
{nodes: Array(0), links: Array(0)}
```

**Comment by n3a9** [comment]
Yes, the idea is that it currently starts as empty, and then calls the API to start populating the arrays. On every new API call and update on nodes/links array is where the issue lies

**Comment by vasturiano** [comment]
I don't see anything being rendered on that codepen example. Is there a way you could put together a reduced example that focuses on just reproducing your issue?

**Comment by n3a9** [comment]
Ah shoot, I changed the api link. Let me see if I can reproduce the error in a new codepen.
